tensorflow/python/ops/math_ops.py

```diff
@@ -99,9 +99,49 @@
 from tensorflow.python.util.tf_export import tf_export
 
 # Aliases for some automatically-generated names.
-linspace = gen_math_ops.lin_space
+#linspace = gen_math_ops.lin_space
 nextafter = gen_math_ops.next_after
 
+def linspace(start_in, stop_in, num, axis=0, name=None):
+  if not isinstance(axis, int):
+    raise TypeError('Axis should be an integer. Received an object {0} of type {1}'.format(axis, type(axis)))
+  with ops.name_scope(name, 'linspace', [start_in, stop_in]):
+    start = ops.convert_to_tensor(start_in, name='start')
+    # stop must be convertible to the same dtype as start
+    stop = ops.convert_to_tensor(stop_in, name='stop', dtype=start.dtype)
+    expanded_start = array_ops.expand_dims(start, axis=axis)
+    num_int = array_ops.convert_to_int_tensor(num, name='num')
+    num = cast(num_int, dtype=start.dtype)
+    shape = expanded_start.get_shape()
```

Comment:
This won't work if expand_start has a partially unknown shape (use tf.shape(expand_start) instead of expand_start.get_shape)

Reply:
Thanks! I added tests for this case - and changed it to use the runtime shape. I don't think we can support fully unknown shape though, we need to know the number of dimensions when the graph is being created because otherwise the `range` later fails.